### DISCUSSION TITLE
Releasing packages using nx_monorepo

### DISCUSSION THREAD

**Discussion by pchaganti** [aws/aws-pdk#245] [Q&A] [answered]
Trying to use nx.monorepo for a repo with a large number of cdk constructs. Any pointers on how to set it up to release all packages with the same version number when changes are pushed to the repo? The changes can be pushed to only one package folder, but would like to release all packages with the same version number. We want all package versions to match a repo release version.

Thanks!

**Comment by JeremyJonas** [comment, marked as answer]
Hi @pchaganti, what your referring to is [lerna fixed/locked mode](https://lerna.js.org/docs/features/version-and-publish#fixedlocked-mode-default) versioning strategy.

Currently the nx-monorepo doesn't manage the sub-package versioning, that is handled by the `bump` tasks from within sub-packages in PDKProject as part of `release` task.

https://github.com/aws/aws-prototyping-sdk/blob/f4f5401f37aa0a81918114bf4c62560c4636e5b0/private/pdk-project.ts#L176-L215

@agdimech is working on new versioning strategy for PDK that would move to pure conventional SemVer versioning, and I believe that is planning to utilize lerna versioning utils. The plan is to still use [lerna independent versioning](https://lerna.js.org/docs/features/version-and-publish#independent-mode), which won't address your question directly. But once we move to lerna it the workspace strategy (fixed vs independent) should become configurable - and we will take this use case into consideration during out refactor.

In the meantime - you will need to implemented this fixed strategy within your workspace by overwriting the sub-packages. You could consider extending the `synth` functionality to overwrite the `bump` task of all sub-project to have the version value from workspace.

https://github.com/aws/aws-prototyping-sdk/blob/f4f5401f37aa0a81918114bf4c62560c4636e5b0/packages/nx-monorepo/src/nx-monorepo.ts#L400-L430

We are definitely open to PRs!
